codec-http/src/main/java/io/netty5/handler/codec/http/headers/HttpHeaders.java

```diff
@@ -0,0 +1,631 @@
+     * @param valuesItr the values of the header.
+     * @return {@code this}.
+     */
+    default HttpHeaders add(CharSequence name, Iterator<? extends CharSequence> valuesItr) {
+        while (valuesItr.hasNext()) {
+            add(name, valuesItr.next());
+        }
+        return this;
+    }
+
+    /**
+     * Adds new headers with the specified {@code name} and {@code values}. This method is semantically equivalent to:
+     *
+     * <pre>
+     * for (T value : values) {
+     *     headers.add(name, value);
+     * }
+     * </pre>
+     *
+     * @param name the name of the header.
+     * @param values the values of the header.
+     * @return {@code this}.
+     */
+    HttpHeaders add(CharSequence name, CharSequence... values);
+
+    /**
+     * Adds all header names and values of {@code headers} to this object.
+     *
+     * @param headers the headers to add.
+     * @return {@code this}.
+     */
+    HttpHeaders add(HttpHeaders headers);
+
+    /**
+     * Sets a header with the specified {@code name} and {@code value}. Any existing headers with the same name are
+     * overwritten.
+     *
+     * @param name the name of the header.
+     * @param value the value of the header.
+     * @return {@code this}.
+     */
+    HttpHeaders set(CharSequence name, CharSequence value);
+
+    /**
+     * Sets a new header with the specified {@code name} and {@code values}. This method is equivalent to:
+     *
+     * <pre>
+     * headers.remove(name);
+     * for (T value : values) {
+     *     headers.add(name, value);
+     * }
+     * </pre>
+     *
+     * @param name the name of the header.
+     * @param values the values of the header.
+     * @return {@code this}.
+     */
+    HttpHeaders set(CharSequence name, Iterable<? extends CharSequence> values);
+
+    /**
+     * Sets a new header with the specified {@code name} and {@code values}. This method is equivalent to:
+     *
+     * <pre>
+     * headers.remove(name);
+     * while (valueItr.hasNext()) {
+     *     headers.add(name, valueItr.next());
+     * }
+     * </pre>
+     *
+     * @param name the name of the header.
+     * @param valueItr the values of the header.
+     * @return {@code this}.
+     */
+    default HttpHeaders set(CharSequence name, Iterator<? extends CharSequence> valueItr) {
+        remove(name);
+        while (valueItr.hasNext()) {
+            add(name, valueItr.next());
+        }
+        return this;
+    }
+
+    /**
+     * Sets a header with the specified {@code name} and {@code values}. Any existing headers with this name are
+     * removed. This method is equivalent to:
+     *
+     * <pre>
+     * headers.remove(name);
+     * for (T value : values) {
+     *     headers.add(name, value);
+     * }
+     * </pre>
+     *
+     * @param name the name of the header.
+     * @param values the values of the header.
+     * @return {@code this}.
+     */
+    HttpHeaders set(CharSequence name, CharSequence... values);
```

Comment:
We could add a default method here by call `set(CharSequence name, Iterator<? extends CharSequence> valueItr)`